Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM golang:1.19 as go
+
+RUN ["mkdir", "-p", "/test"] 
+RUN ["mkdir", "-p", "/test/bin"] 
+ADD features /test/features
+ADD go.mod /test
+ADD go.sum /test
+ADD e2e.go /test
```

Comment:
Why not just `ADD . /test`? If there's stuff you want to skip could use a `.dockerignore` file?

Reply:
Also, you could re-order these to something like:
```Dockerfile
ADD go.mod go.sum /test
RUN go mod download
```
Then `ADD` in everything else, after that.
That way it can cache all the go deps and only rebuild them if the `go.mod` or `go.sum` change.

So overall I'd re-order this block as:
```Dockerfile
RUN ["mkdir", "-p", "/test"] 
RUN ["mkdir", "-p", "/test/bin"] 

# Add the deps
ADD go.mod go.sum /test/
WORKDIR /test
RUN go mod download

# Add the source
ADD . /test/
```

Reply:
ah, good optimizations, incorporated it here - https://github.com/stellar/system-test/pull/1/commits/f8f43513c1f9386d34fd50c7786c182fe044b87f